File: src/chain-observers/classifiers/block-classifier.ts
import type { transaction } from "@hiveio/wax";
import { CollectorClassifierBase } from "./collector-classifier-base";

export interface ITransactionData {
  transaction: transaction;
  id: string;
}

export interface IBlockData {
  transactions: ITransactionData[];
  transactionsPerId: Map<string, transaction>;
}

export class BlockClassifier extends CollectorClassifierBase<IBlockData> {

}

File: python/workerbee/chain_observers/classifiers/block_classifier.py
from __future__ import annotations

from ..classifier_results import BlockClassifierData
from .collector_classifier_base import CollectorClassifierBase


class BlockClassifier(CollectorClassifierBase[BlockClassifierData]):
    """IBlockData:
    transactions: list[ITransactionData]  — each has {transaction, id}
    transactions_per_id: dict[str, transaction]
    """

    pass
